src/test/java/de/retest/suite/flow/ApplyChangesToStatesFlowTest.java

```diff
@@ -78,4 +80,66 @@ public void apply_should_apply_changes_to_states() throws Exception {
 
 		// even more advanced: check if difference between before and after is acceptedChanges!
 	}
+
+	@Test
+	public void apply_should_not_fail_immediatly_when_a_gm_cannot_be_found() throws Exception {
+		final PersistenceFactory persistenceFactory =
+				new PersistenceFactory( new HashSet<>( Arrays.asList( StdXmlClassesProvider.getXmlDataClasses() ) ) );
+		final Persistence<SutState> persistence = persistenceFactory.getPersistence();
+
+		final File originalFile = new File( FILE );
+		final File temporaryFile1 = temp.newFile( "check_GUI_with_review_license.launch1.recheck" );
+		final File temporaryFile2 = temp.newFile( "check_GUI_with_review_license.launch2.recheck" );
+		final File temporaryFile3 = temp.newFile( "check_GUI_with_review_license.launch3.recheck" );
+
+		FileUtils.copyFile( originalFile, temporaryFile1 );
+		final SutState before1 = persistence.load( temporaryFile1.toURI() );
+
+		FileUtils.copyFile( originalFile, temporaryFile2 );
+		final SutState before2 = persistence.load( temporaryFile2.toURI() );
+
+		FileUtils.copyFile( originalFile, temporaryFile3 );
+		final SutState before3 = persistence.load( temporaryFile3.toURI() );
+
+		final SuiteChangeSet acceptedChanges =
+				buildChanges( temporaryFile1, temporaryFile2, temporaryFile3, before1, before2, before3 );
+
+		temporaryFile1.delete();
+		temporaryFile3.delete();
+
+		ApplyChangesToStatesFlow.apply( persistence, acceptedChanges );
+
+		final SutState after2 = persistence.load( temporaryFile2.toURI() );
+		final Element changed2 = after2.getRootElements().get( 0 ).getContainedElements().get( 0 )
+				.getContainedElements().get( 0 ).getContainedElements().get( 0 ).getContainedElements().get( 0 )
+				.getContainedElements().get( 0 );
+
+		assertThat( changed2.getAttributes().get( "enabled" ) ).isEqualTo( true );
+	}
+
+	private SuiteChangeSet buildChanges( final File temporaryFile1, final File temporaryFile2,
```

Comment:
Now you only use this method in the new test case, right? Instead of doing the same thing three times, have a single method and reuse it. Ideally, you also reuse it in the other test case. Remember [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself).